.github/workflows/ci-cd-pipeline.yml

```diff
@@ -1,88 +1,74 @@
-    name: Run API Tests
+    name: API Tests
     steps:
-      - name: Checkout API repository
-        uses: actions/checkout@v2
+      - name: Checkout API repo
+        uses: actions/checkout@v4
         with:
-          repository: 'AlibekNamazbayev/APIAutomation'
+          repository: AlibekNamazbayev/APIAutomation
           token: ${{ secrets.GITHUB_TOKEN }}
-        
-      - name: Setup .NET
-        uses: actions/setup-dotnet@v3
+          path: api-repo
+
+      - name: Setup .NET 9
+        uses: actions/setup-dotnet@v4
         with:
-          dotnet-version: '9.0.x'
+          dotnet-version: | 
+            9.0.x
+            8.0.x  
 
       - name: Restore dependencies
-        run: dotnet restore APIAutomation.sln
-
-      - name: Build API project
-        run: dotnet build APIAutomation.sln --configuration Release
-
-      - name: Run API tests
-        run: dotnet test APITest.Tests/ApiTest.Tests.csproj --logger "trx"
-
-      - name: Upload API test results
-        uses: actions/upload-artifact@v2
-        with:
-          name: api-test-results
-          path: **/*.trx
+        run: dotnet restore ApiTestSolution.sln
+        working-directory: ./api-repo
 
   ui-tests:
     runs-on: ubuntu-latest
-    name: Run UI Tests
+    name: UI Tests
     needs: api-tests
     steps:
-      - name: Checkout UI repository
-        uses: actions/checkout@v2
+      - name: Checkout UI repo
+        uses: actions/checkout@v4
         with:
-          repository: 'AlibekNamazbayev/.NET-Web-UI-Test-Automation-Tools-Part-2'
+          repository: AlibekNamazbayev/.NET-Web-UI-Test-Automation-Tools-Part-2
           token: ${{ secrets.GITHUB_TOKEN }}
+          path: ui-repo
 
       - name: Setup .NET
-        uses: actions/setup-dotnet@v3
+        uses: actions/setup-dotnet@v4
         with:
-          dotnet-version: '9.0.x'
+          dotnet-version: | 
+            9.0.203  
+            8.0.301  
 
-      - name: Restore dependencies
-        run: dotnet restore UIAutomation.sln
-
-      - name: Build UI project
-        run: dotnet build UIAutomation.sln --configuration Release
+      - name: Verify .NET version
+        run: dotnet --info
 
-      - name: Run UI tests
-        run: |
-          dotnet test UIAutomation.Tests/UIAutomation.Tests.csproj --browser ${{ github.event.inputs.browser }} --logger "trx"
+      - name: Clean project
+        run: dotnet clean SeleniumTests.csproj --framework net9.0
+        working-directory: ./ui-repo
 
-      - name: Upload UI test results
-        uses: actions/upload-artifact@v2
-        with:
-          name: ui-test-results
-          path: **/*.trx
+      - name: Restore dependencies
+        run: dotnet restore SeleniumTests.csproj
+        working-directory: ./ui-repo
 
-      - name: Upload screenshots and logs
-        uses: actions/upload-artifact@v2
-        with:
-          name: ui-screenshots
-          path: '**/*.png'
+      - name: Build project
```

Comment:
u can combine building and restoring in 1 step, cause building is making the restoring of nuget packages (without  --no-restore parameter)